src/vm/threadsuspend.cpp

```diff
@@ -8487,12 +8487,12 @@ bool Thread::InjectGcSuspension()
     return false;
 }
 
-#endif // FEATURE_HIJACK && PLATFORM_UNIX
+#endif // !_TARGET_ARM64 && FEATURE_HIJACK && PLATFORM_UNIX
 
 // Initialize thread suspension support
 void ThreadSuspend::Initialize()
 {
-#if defined(FEATURE_HIJACK) && defined(PLATFORM_UNIX)
+#if defined(FEATURE_HIJACK) && defined(PLATFORM_UNIX) && defined(_TARGET_AMD64_)
```

Comment:
We don't need to add the AMD64 ifdef here. The injection should be implemented for all non-windows systems.